crates/interfaces/src/db.rs

```diff
@@ -1,38 +1,39 @@
+use std::fmt::Display;
 use thiserror::Error;
 
 /// Database error type.
 #[derive(Clone, Debug, PartialEq, Eq, Error)]
 pub enum DatabaseError {
     /// Failed to open the database.
-    #[error("failed to open the database ({0})")]
-    Open(i32),
+    #[error("failed to open the database: {0}")]
+    Open(Box<DatabaseErrorInfo>),
```

Comment:
If this weren't in a `Box` (and `::Write` too), `DatabaseError` aligned size would be 80 bytes, instead of 16. `ProviderError` too.

Reply:
IMO no `Box` here is fine: [the lint rule](https://rust-lang.github.io/rust-clippy/master/#/large_enum_variant) is set to the 200 bytes threshold by default, and without `Box` it looks much cleaner